### DISCUSSION TITLE
Cant complete customization

### DISCUSSION THREAD

**Discussion by A3Bagged** [squidfunk/mkdocs-material#7768] [Q&A] [answered]
Hello, i'm trying to follow the [Customization](https://squidfunk.github.io/mkdocs-material/customization/#theme-development) tutorial for `insiders` but once i get to **_Development mode_** it throws me some errors.
_I've cloned the repo from the directory itself since using Guthub Desktop said `fatal: protocol 'https' is not supported` using [git clone with GH_TOKEN](https://squidfunk.github.io/mkdocs-material/customization/#environment-setup-insiders)

**Powershell 7:** _npm start_
```
(venv) PS D:\Documenten\GitHub\mkdocs-material-insiders> npm start

> mkdocs-material@9.5.47+insiders-4.53.14 start
> ts-node -T tools/build --verbose --all --dirty --watch

18:30:04 + material/overrides\assets\stylesheets\custom.css
18:30:04 + material/templates\assets\stylesheets\palette.css
18:30:04 + material/overrides\assets\stylesheets\home.css
Error: EPERM: operation not permitted, symlink '..\..\src\extensions\emoji.py' -> 'D:\Documenten\GitHub\mkdocs-material-insiders\material\extensions\emoji.py'
    at async Object.symlink (node:internal/fs/promises:1001:10) {
  errno: -4048,
  code: 'EPERM',
  syscall: 'symlink',
  path: '..\\..\\src\\extensions\\emoji.py',
  dest: 'D:\\Documenten\\GitHub\\mkdocs-material-insiders\\material\\extensions\\emoji.py'

Error: EPERM: operation not permitted, symlink '..\..\..\src\overrides\hooks\shortcodes.py' -> 'D:\Documenten\GitHub\mkdocs-material-insiders\material\overrides\hooks\shortcodes.py'
    at async Object.symlink (node:internal/fs/promises:1001:10) {
  errno: -4048,
  code: 'EPERM',
  syscall: 'symlink',
  path: '..\\..\\..\\src\\overrides\\hooks\\shortcodes.py',
  dest: 'D:\\Documenten\\GitHub\\mkdocs-material-insiders\\material\\overrides\\hooks\\shortcodes.py'
```

I've tried adding these manually from he github but they get removed everytime.

<hr>

Next i tried to see if it would run at all using `mkdocs serve` in a fresh **_Powershell 7_** window and i got the following
```
PowerShell 7.4.6
PS C:\Users\Christoph> cd D:\Documenten\GitHub\mkdocs-material-insiders
PS D:\Documenten\GitHub\mkdocs-material-insiders> mkdocs serve
ERROR   -  Config value 'markdown_extensions': Failed to load extension 'material.extensions.preview'.
           ModuleNotFoundError: No module named 'material.extensions.preview'

Aborted with a configuration error!
PS D:\Documenten\GitHub\mkdocs-material-insiders>
```

I can't get the preview to work, any ideas?

**Comment by squidfunk** [comment]
I expect this issue to be a problem with Windows, but I don't have Windows at my disposal. Everything works as expected on macOS / Unix. Maybe somebody with access to Windows can help here. The second error might be caused by the fact that the build is not completing. Try to run it on a fresh checkout – all files are checked in. If that works, and you followed all steps in the customization guide, it's related to the symlink problem.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @A3Bagged,

> (venv) PS D:\Documenten\GitHub\mkdocs-material-insiders> npm start

The first issue shows errors with symlink creation. You need to run PowerShell with Administrator privileges or [enable Developer mode](https://learn.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development), I run as admin personally.

> PS D:\Documenten\GitHub\mkdocs-material-insiders> mkdocs serve
> ModuleNotFoundError: No module named 'material.extensions.preview'

Second issue looks like you haven't enabled the `venv` with the Insiders version of the project, hence the lack of the preview module.
You can check `pip show mkdocs-material --verbose`

As for the issue with the https protocol, I don't know, I never used GitHub Desktop 🤔 . If you're planning to work with Insiders and perhaps plan to use the `social` plugin locally, better switch to MSYS2 environment with installed `gcc`, `git` etc. 
- https://squidfunk.github.io/mkdocs-material/plugins/requirements/image-processing/#cairo-graphics-windows

**Reply by kamilkrzyskow** [reply]
Is this resolved @A3Bagged?

**Reply by A3Bagged** [reply]
> Is this resolved @A3Bagged?

Sorry i havent tried this yet as im working on a different custom theme before i got 'insiders' as i dont think the insiders will work anymore when it runs out. Could try this im a bit

**Reply by kamilkrzyskow** [reply]
After you cancel / the subscription runs out, If you create a hard fork copy you can still use the latest available version to you:
- https://squidfunk.github.io/mkdocs-material/insiders/runtime-and-cancellation/#effects-of-cancellation

You won't get further updates, though. Also be sure not to post the source anywhere publicly and keep it private 😎

**Reply by A3Bagged** [reply]
> Is this resolved @A3Bagged?

It worked, turns out i had to run Administrator mode indeed. Thanks!
It's connected on `http:///localhost:8000/mkdocs-material/`

it did throw some errors about `no library called "cairo-2" was found` along with `"cairo"` and `"libcairo-2"` but that's fixable through the link it gave me on the troubleshooting guide

Edit:
i did the troubleshooting fix for cairo2.
Trying the `npm run build` command i get an error about an icon
![image](https://github.com/user-attachments/assets/33dccf14-cd6e-4eef-a378-eb04a03574ab)
